clients/src/test/java/org/apache/kafka/clients/producer/MockProducerTest.java

```diff
@@ -70,185 +81,170 @@ public void testPartitioner() throws Exception {
         PartitionInfo partitionInfo1 = new PartitionInfo(topic, 1, null, null, null);
         Cluster cluster = new Cluster(null, new ArrayList<Node>(0), asList(partitionInfo0, partitionInfo1),
                 Collections.<String>emptySet(), Collections.<String>emptySet());
-        try (MockProducer<String, String> producer = new MockProducer<>(cluster, true, new DefaultPartitioner(), new StringSerializer(), new StringSerializer())) {
-            ProducerRecord<String, String> record = new ProducerRecord<>(topic, "key", "value");
-            Future<RecordMetadata> metadata = producer.send(record);
-            assertEquals("Partition should be correct", 1, metadata.get().partition());
-            producer.clear();
-            assertEquals("Clear should erase our history", 0, producer.history().size());
-        }
+        MockProducer<String, String> producer = new MockProducer<>(cluster, true, new DefaultPartitioner(), new StringSerializer(), new StringSerializer());
```

Comment:
@hachikuji The only anomaly is here, where the producer type is different. This made me use an additional `this.producer != null` check in the `@After` method. Not sure if there is a cleaner way to handle it.